airflow/dag/serialization.py

```diff
@@ -0,0 +1,316 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""DAG serialization with JSON."""
+
+from enum import Enum
+from enum import unique
+import json
+import logging
+import datetime
+import dateutil.parser
+import pendulum
+
+from airflow.models import BaseOperator
+from airflow.models import DAG
+from airflow.models.connection import Connection
+from airflow.www.utils import get_python_source
+
+
+# Serialization failure returns 'failed'.
+FAILED = 'serialization_failed'
+
+
+# Fields of an encoded object in serialization.
+@unique
+class Encoding(str, Enum):
+    """Enum of encoding constants."""
+    TYPE = '__type'
+    VAR = '__var'
+
+
+# Supported types for encoding. primitives and list are not encoded.
+@unique
+class DagTypes(str, Enum):
```

Comment:
I think we should change it to something like below as `DagType` sounds something different. WDYT? 

```suggestion
class DagAttributeTypes(str, Enum):
```

Suggesting this just for clarity - don't have a strong preference on it.

Reply:
It makes sense. I changed it.